sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client.py

```diff
@@ -3741,3 +3745,39 @@ def seal_append_blob(self, **kwargs):
             return self._client.append_blob.seal(**options) # type: ignore
         except HttpResponseError as error:
             process_storage_error(error)
+
+    @distributed_trace
+    def get_container_client(self, **kwargs):
+        # type: (...) -> ContainerClient
+        """Get a client to interact with the blob's parent container.
+
+        The container need not already exist. Defaults to current blob's credentials.
+
+        :keyword credential:
+            This enables you to change credentials when its necessary. The value can be a SAS token string,
+            an instance of a AzureSasCredential from azure.core.credentials, an account shared access
+            key, or an instance of a TokenCredentials class from azure.identity.
+        :returns: A ContainerClient.
+        :rtype: ~azure.storage.blob.ContainerClient
+
+        .. admonition:: Example:
+
+            .. literalinclude:: ../samples/blob_samples_containers.py
+                :start-after: [START get_container_client_from_blob_client]
+                :end-before: [END get_container_client_from_blob_client]
+                :language: python
+                :dedent: 8
+                :caption: Get container client from blob object.
+        """
+        from ._container_client import ContainerClient
+        _pipeline = Pipeline(
+            transport=TransportWrapper(self._pipeline._transport), # pylint: disable = protected-access
```

Comment:
I'm not sure I follow, if a user gets a container client from a bsc we wrap and then if we get a blob client from that container client then we wrap again - isn't the same as this case except we're just going upwards rather than downwards?

Reply:
oh I didn't paste the link successfully https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/storage/azure-storage-blob/azure/storage/blob/_shared/authentication.py#L81-L83
if you run test_create_blob_with_equal_sign and put debug point on line 81 probably you would know what I mean

Reply:
OH haha I see why it works in the case of bsc -> cc -> bc 
Hmmm the problem is if we don't wrap there will be issues with context managers - I think. I will investigate this more

Reply:
I have updated the PR, the methods now create a newly constructed pipeline (and transport) - this means that their context is independent from the child class that created them. 
The other approach was to pass the child class transport object (without wrapping)  - this means that in context managers would get closed if the child got closed (if the child was not wrapped) and would always remain open if the child class was wrapped (since child classes only close when their parent closes)

I felt the best approach would just be to create a new pipeline when these constructors are used. Let me know what you think

Reply:
By creating new pipeline and transport object, we will not be able to reuse the existing connection pool.
Can we check if it's a transportwrapper and if it's already a transport wrapper we can directly reuse the pipeline? by specifying self._pipeline.

Reply:
I thought of this before, my concern with this is that if we spin up BSC from CC for example, then if CC is closed the BSC will get closed as well. Is this ok? - As I'm writing this I realized that its ok to do that actually